factory/base.py

```diff
@@ -733,7 +733,7 @@ def _build(cls, model_class, *args, **kwargs):
             raise ValueError(
                 "ListFactory %r does not support Meta.inline_args.", cls)
 
-        values = [v for k, v in sorted(kwargs.items())]
+        values = [v for k, v in sorted(kwargs.items(), key=lambda item: int(item[0]))]
```

Comment:
Maybe add a comment to change this to `values = kwargs.values()` when dropping support for Python < 3.6?